src/main/kotlin/org/malachite/estella/offer/api/OfferController.kt

```diff
@@ -29,50 +29,47 @@ class OfferController(
     fun getOffers(): ResponseEntity<List<OfferResponse>> =
         offerService.getOffers()
             .map { it.toOfferResponse() }
-            .let { ResponseEntity(it, HttpStatus.OK) }
+            .let { ResponseEntity.ok(it) }
 
 
     @CrossOrigin
     @GetMapping("/{offerId}")
     fun getOffer(@PathVariable offerId: Int): ResponseEntity<OfferResponse> =
         offerService.getOffer(offerId)
-            .let { ResponseEntity(it.toOfferResponse(), HttpStatus.OK) }
+            .let { ResponseEntity.ok(it.toOfferResponse()) }
 
     @CrossOrigin
     @PostMapping()
     fun addOffer(
         @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
         @RequestBody offer: OfferRequest
-    ): ResponseEntity<Message> {
-        val hrPartner = getHrPartnerFromJWT(jwt)
-        offerService.addOffer(offer, hrPartner)
-        return ResponseEntity(SuccessMessage, HttpStatus.OK)
-    }
+    ): ResponseEntity<OfferResponse> =
+        getHrPartnerFromJWT(jwt)
+            .let { offerService.addOffer(offer, it) }
+            .let(Offer::toOfferResponse)
+            .let { OwnResponses.CREATED(it) }
 
     @CrossOrigin
     @PutMapping("/{offerId}")
     fun updateOffer(
         @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
         @PathVariable("offerId") offerId: Int,
         @RequestBody offer: OfferRequest
-    ): ResponseEntity<Any> {
-        val hrPartner = getHrPartnerFromJWT(jwt)
-        offerService.updateOffer(offerId, offer, hrPartner)
-        return ResponseEntity(SuccessMessage, HttpStatus.OK)
-    }
+    ): ResponseEntity<Any> =
+        getHrPartnerFromJWT(jwt)
+            .let { offerService.updateOffer(offerId, offer, it) }
+            .let { OwnResponses.SUCCESS }
 
     @CrossOrigin
     @DeleteMapping("/{offerId}")
     fun deleteOffer(
         @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
         @PathVariable("offerId") offerId: Int
-    ): ResponseEntity<Message> {
+    ): ResponseEntity<Any> =
         getHrPartnerFromJWT(jwt)
```

Comment:
Idealnie by było napisać to analogicznie jak jest w `applyForAnOffer` z userem w `ApplicationController` żeby było spójnie.

Reply:
Kontroler aplikacji jeszcze będzie refactorowany, więc to na spokojnie się tam wtedy poprawi

Reply:
@tomros766 tobie chodziło żeby tutaj też był UNATUH czy co, bo nie do końca rozumiem XD